nautobot/docs/user-guide/administration/upgrading/from-v1/ipam-upgrade-guide.md

```diff
@@ -0,0 +1,191 @@
+
+In Nautobot 2.0, creating or updating prefixes that violate this guidance will result in a warning; in a future Nautobot release this will be changed to an enforced data constraint.
+
+### Prefix.is_pool field and "Container" status replaced by new field Prefix.type
+
+A new type field was added to `Prefix` to replace the `is_poo`l boolean field and the "Container" status. The `type` field can be set to "Network", "Pool" or "Container", with "Network" being the default.
+
+Existing prefixes with a status of "Container" will be migrated to the "Container" `type`. Existing prefixes with `is_pool` set is migrated to the "Pool" `type`. Prefixes with both `is_pool set` and a status of "Container" are migrated to the "Pool" `type`.
+
+The "Container" status will be removed and all prefixes will be migrated to the "Active" status if it exists. If the"Active" status was deleted, prefixes will be migrated to the first available prefix status in the database that is not"Container".
+
+### Prefix utilization calculations were revamped
+
+The `get_utilization` method on the `ipam.Prefix` model has been updated in 2.0 to account for the `Prefix.type` field as described above under **Prefix Parenting Guidance**. The behavior is now as follows:
+
+- If the `Prefix.type` is `Container`, the utilization is calculated as the sum of the total address space of all child prefixes.
+- If the `Prefix.type` is `Pool`, the utilization is calculated as the sum of the total number of IP addresses within the pool's range.
+- If the `Prefix.type` is `Network`:
+    - The utilization is calculated as the sum of the total address space of all child `Pool` prefixes plus the total number of child IP addresses.
+    - For IPv4 networks with a `prefix_length` larger (lower) than `/31`, if neither the first or last address is occupied by either a pool or an IP address, they are subtracted from the total size of the prefix.
+
+#### Example
+
+- 192.168.0.0/16          `Container - 1024/65536 utilization`
+    - 192.168.1.0/24      `Network - 1/254 utilization`
+        - 192.168.1.1     `IP Address`
+    - 192.168.2.0/24      `Network - 4/256 utilization`
+        - 192.168.2.0/30  `Pool - 1/4 utilization`
+            - 192.168.2.1 `IP Address`
+    - 192.168.3.0/24      `Network - 5/254 utilization`
+        - 192.168.3.1     `IP Address`
+        - 192.168.3.64/30 `Pool - 0/4 utilization`
+    - 192.168.4.0/24      `Network - 1/256 utilization`
+        - 192.168.4.255   `IP Address`
+
+### IPAddress Parenting Concrete Relationship was added
+
+The `ipam.IPAddress` model has been modified to have a mandatory foreign key to `ipam.Prefix` as the `parent` field. Parenting of IP addresses is now automatically managed at the database level to greatly improve performance especially when calculating tree hierarchy and utilization.
+
+| Removed                | Replaced With   |
+| ---------------------- | --------------- |
+| `get_child_prefixes()` | `descendants()` |
```

Comment:
This is a Prefix API, not an IPAddress API, right? Seems out of place in this section.

Reply:
Hmm yeah weird copy pasta. I suppose.